### ISSUE TITLE
Improve UsdPreviewSurface documentation and support

### ISSUE THREAD

**Issue by kwokcb** [AcademySoftwareFoundation/MaterialX#1060]
This issue to to add clarity of handling normal and colorspace inputs / outputs for `UsdPreviewSurface` and `UsdUvTexture`:

# Normal Handling
The normal input on `UsdPreviewSurface` can be confusing to users as there is no proper indication of what values are expected and how to set up input mappers to route the information correctly.

## Normal Input Documentation
Documentation can include the following information 
     * The input is normal offsets in tangent space between the { -1,1,1 } -> { 1,1,1 } range.
     * An appropriate scale and offset is applied within the definition of UsdPreviewSurface and then a remapping back to world space via a normalmap node. 
        * Normal scaling on normalmap currently is not accessible.
     * The default input is { 0,0,1 } which allows for no transformation to be performed. This differs from the default for tangent space `normalmap` input which is { 0.5, 0.5, 1 }.
      
## Normal Map Input
UsdUVTexture has only a color variant. The specification relies on knowledge of the input which determines the output type. It is not possible to specify this in a MaterialX definition especially as resource mapping is not part of the definition. 
     * **The proposal is to add in  a vector3 variant** to allow normal mapping to be supported. 
     * A scale and bias of : `* 2 - 1` is required in the vector3 variant. 
     * The default fallback value can be { 0.5, 0.5, 1.0 } to be consistent with the current `<normalmap>` definition.

# Colorspace handling
The UsdUVTexture specification denotes a "source color space" ("sourcecolorspace") input. 
   * This differs from the convention used for MaterialX. 
   * The proposal is to not add this in this new input but continue to just support "colorspace" input tagging on the filename input as with the existing `image` and `tiledimage` nodes.
   * Mapping from `colorspace` to `sourcecolorspace` is externally defined / implemented as required. 
   
---
In summary:
1. Add in better documentation for the normal input on `UsdPreviewSurface` so that users understand what is required as an entry point. 
2. Be able to add in a vector3 version of `UsdUvTexture` so that a normalmap can be routed to this input.
2a. It would also be useful to document how / add a test to use a image + normalmap node configuration.
3. Propose to only support `colorspace` tagging on the filename attribute for `UsdUvTexture` and not add in a new "sourcecolorspace" input.

**Comment by jstone-lucasfilm** [comment]
Given the improvements to the documentation of the UsdPreviewSurface interface in recent years, perhaps we should close this broader task and write up new tasks for more specific improvements that are still needed?

https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/libraries/bxdf/usd_preview_surface.mtlx

**Comment by jstone-lucasfilm** [comment]
Let's close out this issue as proposed above, and we can write up new tasks for more specific improvements that would still be valuable.
